src/scheduler/scheduler.ts

```diff
@@ -0,0 +1,183 @@
+    ExactlyOncePerInterval = 'ExactlyOncePerInterval',
+}
+
+export class SchedulerConfig {
+    crontab ?: string = '* * * * *'; // Every minute
+    mode ?: SchedulerConcurrencyMode = SchedulerConcurrencyMode.ExactlyOncePerInterval;
+}
+
+////
+// Method Decorator
+////
+
+// Scheduled Time. Actual Time, number running globally, number running locally
+export type ScheduledArgs = [Date, Date, number, number]
+
+export interface SchedulerRegistrationConfig {
+    schedulerConfig?: SchedulerConfig;
+}
+
+export class SchedulerRegistration<This, Args extends unknown[], Return> extends MethodRegistration<This, Args, Return>
+    implements SchedulerRegistrationConfig
+{
+    schedulerConfig?: SchedulerConfig;
+    constructor(origFunc: (this: This, ...args: Args) => Promise<Return>) {
+        super(origFunc);
+    }
+}
+  
+export function Scheduled(schedulerConfig: SchedulerConfig) {
+    function scheddec<This, Ctx extends WorkflowContext, Return>(
+        target: object,
+        propertyKey: string,
+        inDescriptor: TypedPropertyDescriptor<(this: This, ctx: Ctx, ...args: ScheduledArgs) => Promise<Return>>
+    ) {
+        const { descriptor, registration } = registerAndWrapFunction(target, propertyKey, inDescriptor);
+        const schedRegistration = registration as unknown as SchedulerRegistration<This, ScheduledArgs, Return>;
+        schedRegistration.schedulerConfig = schedulerConfig;
+
+        return descriptor;
+    }
+    return scheddec;
+}
+
+///////////////////////////
+// Scheduler Management
+///////////////////////////
+
+export class DBOSScheduler{
+    constructor(readonly dbosExec: DBOSExecutor) {}
+
+    schedLoops: DetachableLoop[] = [];
+    schedTasks: Promise<void> [] = [];
+  
+    initScheduler() {
+        for (const registeredOperation of this.dbosExec.registeredOperations) {
+            const ro = registeredOperation as SchedulerRegistration<unknown, unknown[], unknown>;
+            if (ro.schedulerConfig) {
+                const loop = new DetachableLoop(this.dbosExec, ro.schedulerConfig.crontab ?? '* * * * *', ro);
+                this.schedLoops.push(loop);
+                this.schedTasks.push(loop.startLoop());
+            }
+        }
+    }
+  
+    async destroyScheduler() {
+        for (const l of this.schedLoops) {
+            await l.stopLoop();
+        }
+        this.schedLoops = [];
+        await Promise.all(this.schedTasks);
+        this.schedTasks = [];
+    }
+  
+    logRegisteredSchedulerEndpoints() {
+        const logger = this.dbosExec.logger;
+        logger.info("Scheduled endpoints:");
+        this.dbosExec.registeredOperations.forEach((registeredOperation) => {
+            const ro = registeredOperation as SchedulerRegistration<unknown, unknown[], unknown>;
+            if (ro.schedulerConfig) {
+                logger.info(`    ${ro.name} @ ${ro.schedulerConfig.crontab ?? '* * * * *'}; ${ro.schedulerConfig.mode ?? 'Exactly Once Per Interval'}`);
```

Comment:
Aren't both properties going to be defined, with their default value if none other provided, after having been instantiated in the registration step?

Reply:
I am not thinking so, this being JavaScript.  Constructors do that, but it doesn't get called for sure does it?